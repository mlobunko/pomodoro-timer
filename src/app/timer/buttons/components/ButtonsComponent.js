import React from "react";
import PropTypes from "prop-types";

import Button from "shared/button/components";

import CenterButton from "./CenterButtonContainer";
import { StyledButtonsComponent } from "./styles";

export const ButtonsComponent = ({ openSettings, resetTimer }) => (
  <StyledButtonsComponent>
    <Button side icon="cog" onClick={openSettings} />
    <CenterButton />
    <Button side icon="sync" onClick={resetTimer} />
  </StyledButtonsComponent>
);

ButtonsComponent.displayName = "Buttons";

export default ButtonsComponent;

ButtonsComponent.propTypes = {
  openSettings: PropTypes.func.isRequired,
  resetTimer: PropTypes.func.isRequired
};
