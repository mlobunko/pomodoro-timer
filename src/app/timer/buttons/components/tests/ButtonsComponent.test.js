import React from "react";
import { shallow } from "enzyme";

import { ButtonsComponent } from "../ButtonsComponent";

const props = {
  openSettings: jest.fn(),
  resetTimer: jest.fn()
};

describe("props", () => {
  it("button cog, onClick is 'openSettings'", () => {
    const wrapper = shallow(<ButtonsComponent {...props} />);
    expect(wrapper.find("[icon='cog']").prop("onClick")).toEqual(
      props.openSettings
    );
  });

  it("button sync, onClick is 'resetTimer'", () => {
    const wrapper = shallow(<ButtonsComponent {...props} />);
    expect(wrapper.find("[icon='sync']").prop("onClick")).toEqual(
      props.resetTimer
    );
  });
});

describe("render", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<ButtonsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
