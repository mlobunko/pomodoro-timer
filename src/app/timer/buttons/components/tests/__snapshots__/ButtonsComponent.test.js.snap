// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly 1`] = `
<StyledButtonsComponent>
  <Button
    icon="cog"
    onClick={[Function]}
    side={true}
  />
  <Connect(CenterButton) />
  <Button
    icon="sync"
    onClick={[Function]}
    side={true}
  />
</StyledButtonsComponent>
`;
